### ISSUE TITLE
add-ons store: the description of the add-ons is unreadable with the up and down arrow keys and in say all.

### ISSUE THREAD

**Issue by paulber19** [nvaccess/nvda#17568]
When we use the up and down arrows to read the add-on description, it is read word by word.
When we use "nvda+down arrow" to read the entire add-on description, words are split.
### Actual behavior:

- open add-ons store dialog.
- choose an add-on and press "tab" to socus the description window.
- presse downArrow key to read the description.

Each time you press, NVDA only reads one or two words of the description.

- now, press "control+home" to return to the beginning of the description
- press "NVDA+downArrow" to read all the description.

The description is read but some words are split and spaces are inserted.

For exemple, there is the description that is read for the "Emoticons" add-on:
>>Enables  the  announce ment of  emoticon  names  instead of  the  character  Representation.
The word announcement is split.

For exemple, there is the description that is read for the "Clip Contents Designer" add-on:
>>Allows the  joining of  separate  fragments  of text on  the  clipboard  and  clearing of  clipboard content. 
>>Current  text on  the  clipboard  can be  presented  in browse  mode. 
>>Option to  request  confirmati on before  performin g actions  such as  copy, cut,  add text,  or clearing 
>>the  clipboard  to avoid  accidental  changes.

The words confirmation, performing, are split.

### Expected behavior:

Reading should not be done word by word with upArrow and downArrow keys.
Words should not be broken up into saying all
#### NVDA installed/portable/running from source:
NVDA portable
#### NVDA version:
NVDA version alpha-34834,ada79080 x86
#### Windows version:
Windows 10 22H2 (10.0.19045) workstation AMD64
#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No problem with nvda 2024.4.1
#### If NVDA add-ons are disabled, is your problem still occurring?
The add-ons are necessary.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
No tested.

**Comment by SaschaCowley** [comment]
@paulber19 could you try maximising the Add-on Store window to see if this fixes the problem?

**Comment by paulber19** [comment]
The problem occurs when the window is maximized, but not when it is 
restored.

**Comment by paulber19** [comment]
> Have you reduced your display resolution,
No
> or increased your system text size?
No.

> 2.Could you provide a log in debug mode (and without add-ons)
You need at least one add-on to read its description.
If the log is needed I will provide it but I don't know if attachments 
are accepted.
What I do is quite simple.
I launch the addon store using the NVDA menu.
The window opens, the list of installed add-ons is automatically in 
focus and the first add-on is selected.
By tabbing, I place the focus in the add-on description area.
With the downArrow and uparrow I try to read this description.
Each time the cursor moves, only one or two words are read.
By "alt+space", I check that the window is indeed maximized This is the 
case because the choice to maximize is unavailable.
If using this menu, I choose to restore the window and return to the 
description area, reading is then done correctly.
If the window is maximized again, the problem returns.
I would like to point out that all this does not happen with NVDA 
2024.4.1. Which suggests that it is not the display state of W10 that is 
the problem.


> 3.Could you update the initial description indicating that the Add-on 
> Store window should be maximized? And while at it, move the steps to 
> reproduce the issue in the correct section
Sorry, but I will not embark on this adventure. I'm already having a 
hard time filling out the bug report and I'm afraid of doing something 
stupid.

**Comment by CyrilleB79** [comment]
I mean you provide us a log with add-ons disabled. If I understand correctly your issue, you can still start NVDA with add-ons disabled, but got to the add-on store, select an add-on and read its description.

Regarding attachments, it may not be possible to send them to GitHub if you reply be e-mail. But if you open GitHub's page of this very PR, and write your answer on this page, before sending it, tab twice and you will find a button "Paste, drop, or click to add files". Press it and you will land in a usual open file dialog allowing to select your file. Once done, do not forget to send your answer with the "Comment" button. Not all file types are supported for attachments, but .log are.

If for any reason it's not doable for you, you may send me the log privately and, if you agree, I may post it here myself. If you prefer to share your log only with me, it's also possible; just let me know explicitly.

What you do is probably simple. But given I am not able to reproduce, I am trying, with the log, to find a more reliable and objective way to reproduce, or to understand why I cannot.

At last, regarding the edition of the initial description of this message, it's not mandatory, we have all the information anyway. Though, if you wish me to help you to learn to do this, I can; just ask me. We may discuss it through e-mail in French, this may be easier.

**Comment by SaschaCowley** [comment]
@paulber19 could you also please provide a screenshot of the Add-on Store window in the state that it is in when you are experiencing the issue? If you are unsure if the screenshot contains personal information, you may email it to info@nvaccess.org mentioning this issue.

**Comment by CyrilleB79** [comment]
@paulber19 has sent me the logs:
* the log with the maximized window, where the issue occurs: [NVDA log avec la fenêtre addon store agrandie.txt](https://github.com/user-attachments/files/18353838/NVDA.log.avec.la.fenetre.addon.store.agrandie.txt)
* the log with the restored window, where the issue does not occur: [NVDA log avec fenêtre addon store restaurée.txt](https://github.com/user-attachments/files/18353852/NVDA.log.avec.fenetre.addon.store.restauree.txt)

Unfortunately, I cannot find any issue, nor any difference with what I do in the STR.

Next step would be a visual check as requested by @SaschaCowley.

**Comment by paulber19** [comment]
I just noticed something:
- no screen is connected to my PC.
By plugging in a screen, the problem disappeared.
- when I unplugged the screen, it reappeared.
- in landscape mode, the problem disappeared.
- in portrait mode, he reappeared.

**Comment by Adriani90** [comment]
@paulber19  are all your installed addons disabled?
And does this problem occur also on the available addons tab?

**Comment by CyrilleB79** [comment]
@Adriani90 the log provided by @paulber19 (see https://github.com/nvaccess/nvda/issues/17568#issuecomment-2578826059) clearly indicate that Paul has launched NVDA with add-ons disabled.

The screen orientation or presence seems a good track, even if I do not know why the issue has only appeared recently.

**Comment by paulber19** [comment]
Continuing my research, I noticed that the size of the "description" 
text area of ​​the addon store window varies a lot depending on the 
display mode:

with nvda alpha:
in landscape: location: RectLTWH(left=681, top=88, width=304, height=100)
in portrait: location: RectLTWH(left=681, top=88, width=48, height=100)

In portrait, the width is very small.
Contrary to what I said for NVDA 2024.4, the problem also exists but is 
less sensitive.

The width also varies but to a lesser extent:
with nvda 2024.4:
in landscape: location: RectLTWH(left=527, top=88, width=458, height=100)
in portrait: location: RectLTWH(left=527, top=88, width=202, height=100)

In conclusion, I think it would be useful to ensure that this text box 
has sufficient width to be understandable regardless of the display mode.

**Comment by CyrilleB79** [comment]
The situation has become worse probably due to the introduction of a new column in #17091.

We end up in a situation where we probably cannot have both all the columns of the list fully visible and the description width enough to avoid splitting the text in too small chunks.

**Comment by seanbudd** [comment]
I think its possible to add minimum widths to these controls

**Comment by gerald-hartig** [comment]
This can potentially be solved by making the add-on list horizontally scrollable and by adding a minimum width to the add-on description.

**Comment by valiant8086** [comment]
could also move the description to above or below the list I guess. I think the previous comment is probably the cleanest approach though. Either that or you have to do something to retrieve the description e.g. tab and press an about button or press an accellerator key combination to do it faster, which brings up a popup dialogue box with focus in the readonly and you can escape to return to the add-on store or click ok, etc. If we're doing that, might not hurt to place buttons to do things to add-ons in there. There are obviously other ways to do it, but someone might find it handy to have everything out in the open on this about dialogue. But I have no idea if there's an issue with such a dialogue blocking the main add-on store window thread, etc.
